src/swap/handle_swap_sign_transaction.c

```diff
@@ -119,7 +121,75 @@ bool swap_copy_transaction_parameters(create_transaction_parameters_t* params) {
     return true;
 }
 
-bool swap_check_validity() {
+static address_t* swap_get_tx_recipient_address(void) {
+    address_t* recipient = NULL;
+
+    /*
+     *  XXX:
+     *   In case of Jetton transfer, transaction `to` address is the sender jetton wallet address.
+     *   Recipient address is the second hints in message hints.
+     */
+    if (G_context.tx_info.transaction.hints_type == TRANSACTION_TRANSFER_JETTON) {
+        if (G_context.tx_info.transaction.hints.hints_count >= 2) {
```

Comment:
I think this pattern can be dangerous
The info of whether we are doing jetton or native transfer should only come top down Exchange -> application
The source of truth of the transfer type should be the `strncmp(G_swap_validated.ticker, "TON", sizeof("TON")) == 0)` check below and it should then be forwarded to all checks that need it

Reply:
True